### PR TITLE
Add kotlin code for dynamic programming

### PR DESCRIPTION
**Pull request by curtishd**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.
屏幕截图 2024-02-28 101219
屏幕截图 2024-02-28 101230
屏幕截图 2024-02-28 101239
屏幕截图 2024-02-28 101248
屏幕截图 2024-02-28 101259
屏幕截图 2024-02-28 101311
屏幕截图 2024-02-28 101320
屏幕截图 2024-02-28 101331
屏幕截图 2024-02-28 101341
屏幕截图 2024-02-28 101349
屏幕截图 2024-02-28 101359
屏幕截图 2024-02-28 101409

### CODE REVIEW

File: codes/kotlin/chapter_dynamic_programming/knapsack.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
+/**
+ * File: kt
```
Comment (krahets) [inline codes/kotlin/chapter_dynamic_programming/knapsack.kt:2]: file name

Comment (curtishd) [reply codes/kotlin/chapter_dynamic_programming/knapsack.kt]: fix

Comment (curtishd) [reply codes/kotlin/chapter_dynamic_programming/knapsack.kt]: fixed!

Comment (curtishd) [reply codes/kotlin/chapter_dynamic_programming/knapsack.kt]: Fixed

**Review by krahets — APPROVED** [review]
Thanks for the PR!

**Pull request merged**